Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PaalmanPingsMonteCarloAbsorption.py

```diff
@@ -378,8 +383,16 @@ def _set_beam(self, ws):
                                               'Height': self._beam_height})
         set_beam_alg.execute()
 
-    def _set_sample(self, ws, components, can_as_sample=False):
+    def _set_sample_material_only(self, ws):
+        # this exists for the case in which the user wants to use preexisting sample/can geometries but override the
+        # sample material properties
+        set_sample_alg = self.createChildAlgorithm("SetSample", enableLogging=False)
+        set_sample_alg.setProperty("InputWorkspace", ws)
+        sample_material = self._set_material_dict('sample')
+        set_sample_alg.setProperty("Material", sample_material)
+        set_sample_alg.execute()
 
+    def _set_sample(self, ws, components):
         # this is to ensure that there is no historic remnant of the sample or container that can affect the attenuation
         # from the other hand this does not allow for sample nor container shape nor material to be already defined
```

Comment:
This comment is no longer relevant.

Reply:
I think it may be, as in this use case the user has provided data to create geometries for both the sample and can. Hence the initial call to `setSample` with a blank sample on line 398.